es_source.go

```diff
@@ -87,6 +87,25 @@ func NewElasticsearchSource(config *ElasticsearchConfig) (Source, error) {
 	}, nil
 }
 
+// Create a new ElasticsearchSource from the input object, but add extra features
+// to include to the new ElasticsearchSource instance.
+func (e *ElasticsearchSource) addExtraFeatures(extraFeatures map[string]string) *ElasticsearchSource {
```

Comment:
If I understand this correctly, by "features" here you are referring to extra parameters. I recommend renaming to `addExtraFields` to avoid the confusion with "features" as in "geo features".

Reply:
Agreed. -- "Feature" in Mapbox Speak is some entity in a layer, such as a 3D object. And those have Properties, which we derive from SourceFields here.

I'll do that in a separate commit, since we have to keep things consistent on the caller side as well...